File: ejercicios python 2/ejercicio 3.py
def calculadora_api(operacion=None, num1=None, num2=None):
    while True:
        try:
            operacion = (int(input("ingrese la operacion que desa hacer: \n 1- Suma \n 2- Resta \n 3- Multiplicacion \n 4- Division \n  5- Fibonacci \n Tu opcion: ")))
            if operacion < 0 or operacion > 5:
                print("opcion no valida")
            else:
                break
        except ValueError:
            print("ingrese un numero del 1 al 5")
    
    while True:
        try:
            num1 = (int(input("ingrese valor de num1:")))
            if num1 < 0:
                print("no puede ser numero negativo")
            else:
                break
        except ValueError:
            print("ingrese un numero valido")
            
    while True:
        try:
            num2 = (int(input("ingrese valor de num1:")))
            if num2 < 0:
                print("no puede ser numero negativo")
            else:
                break
        except ValueError:
            print("ingrese un numero valido")

    # -------------------------
    # OPERACIONES BÁSICAS
    # -------------------------
    if operacion == 1:
        resultado = num1 + num2
        print(f"el resultado de la suma entre {num1} + {num2} = {resultado}")

    elif operacion == 2:
        resultado = num1 - num2
        print(f"el resultado de la resta entre {num1} - {num2} = {resultado}")

    elif operacion == 3:
        resultado = num1 * num2
        print(f"el resultado de la multiplicacion entre {num1} * {num2} = {resultado}")

    elif operacion == 4:
        if num2 == 0:
            return "Error: División entre 0"
        resultado = num1 / num2
        print(f"el resultado de la division entre {num1}  {num2} = {resultado}")

    # -------------------------
    # FIBONACCI
    # -------------------------
    elif operacion == 5:
        n = num1
        a, b = 0, 1
        serie = []

        for _ in range(n):
            serie.append(a)
            a, b = b, a + b

        print(serie)

    else:
        print("Operación no válida")

    # -------------------------
    # CONVERSIÓN DE BASES
    # -------------------------
    if resultado is not None:
        print("-" * 20)
        print(f"Resultado Decimal: {resultado}")
        print(f"Binario: {bin(int(resultado))}")
        print(f"Hexadecimal: {hex(int(resultado))}")
        
calculadora_api()

File: ejercicios java semana 2/ejercicio 3.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function preguntar(pregunta) {
    return new Promise(resolve => {
        rl.question(pregunta, respuesta => {
            resolve(respuesta);
        });
    });
}

async function calculadoraAPI() {

    let operacion;

    // -------------------------
    // MENÚ
    // -------------------------
    while (true) {
        let input = await preguntar(
            "Ingrese la operación:\n" +
            "1- Suma\n" +
            "2- Resta\n" +
            "3- Multiplicación\n" +
            "4- División\n" +
            "5- Fibonacci\n" +
            "Tu opción: "
        );

        operacion = parseInt(input);

        if (!isNaN(operacion) && operacion >= 1 && operacion <= 5) {
            break;
        }

        console.log("Opción no válida");
    }

    let num1, num2;

    // -------------------------
    // PEDIR NÚMEROS
    // -------------------------
    while (true) {
        let input = await preguntar("Ingrese valor de num1: ");
        num1 = parseInt(input);

        if (!isNaN(num1) && num1 >= 0) break;

        console.log("Ingrese un número válido (no negativo)");
    }

    while (true) {
        let input = await preguntar("Ingrese valor de num2: ");
        num2 = parseInt(input);

        if (!isNaN(num2) && num2 >= 0) break;

        console.log("Ingrese un número válido (no negativo)");
    }

    let resultado;

    // -------------------------
    // OPERACIONES
    // -------------------------
    if (operacion === 1) {
        resultado = num1 + num2;
        console.log(`Resultado suma: ${resultado}`);
    }

    else if (operacion === 2) {
        resultado = num1 - num2;
        console.log(`Resultado resta: ${resultado}`);
    }

    else if (operacion === 3) {
        resultado = num1 * num2;
        console.log(`Resultado multiplicación: ${resultado}`);
    }

    else if (operacion === 4) {
        if (num2 === 0) {
            console.log("Error: División entre 0");
            rl.close();
            return;
        }

        resultado = num1 / num2;
        console.log(`Resultado división: ${resultado}`);
    }

    else if (operacion === 5) {
        let n = num1;
        let a = 0, b = 1;
        let serie = [];

        for (let i = 0; i < n; i++) {
            serie.push(a);
            [a, b] = [b, a + b];
        }

        console.log("Serie Fibonacci:", serie);
        rl.close();
        return;
    }

    // -------------------------
    // CONVERSIÓN DE BASES
    // -------------------------
    if (resultado !== undefined) {
        console.log("--------------------");
        console.log("Decimal:", resultado);
        console.log("Binario:", Math.trunc(resultado).toString(2));
        console.log("Hexadecimal:", Math.trunc(resultado).toString(16));
    }

    rl.close();
}

// Ejecutar
calculadoraAPI();
